### ISSUE TITLE
[TW-945] regex support

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#980]
_David Patrick on 2009-07-05T00:50:13Z says:_ 

 whatever that entails

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2009-07-05T00:50:13Z
Modified: 2014-02-09T02:11:53Z
```

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2010-08-01T14:21:35Z says:_ 

 Regexes could be supported in filters, like:

<pre>
description.rx:^[A-Za=z]+[0-9]{1,3}
</pre>

I made that up, but the mechanism could be like that.  Is it feasible to support regexes wherever text could be entered?  Or would it be confusing, and require mechanism like above?

<pre>
task list pro:.{1,4}
task 123 /?onday/Tuesday/g
</pre>

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2010-11-26T17:28:19Z says:_ 

 Using rc.regex=on/off (default off) now controls whether filters and substitutions support regular expressions.

No change in syntax, despite my earlier comments.

**Comment by taskwarrior** [comment]
_Max Muller on 2011-04-26T19:28:03Z says:_ 

 Paul, did regex make it into 1.9.4 after all? The changelog for that version says this feature is now implemented. So I put regex=on in my rc file, and when I run the (at)task diagnostics@ command, the resulting report tells me this "Regex: Enabled"

But when I try to use it, regex doesn't seem to work in either filters or substitutions.

Anyone else having a problem with this?

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-04-27T02:38:42Z says:_ 

 Max,

Regex support was implemented in 1.9.4, but pulled during the beta.  There were a lot of bugs - too many, for such a small change.  I want a comprehensive set of unit tests that address searching with and without regexes enabled, but that hasn't been created yet.

The feature is still there, but conditionally compiled out.
